framework/test/jest/integration/specs/modules/chain/components/storage/entities/chain_meta.spec.js

```diff
@@ -0,0 +1,248 @@
+		});
+
+		it('should reject with error if provided filter does not match', async () => {
+			await expect(
+				ChainMetaEntity.getOne({ key: 'custom-key' })
+			).rejects.toThrow('No data returned from the query.');
+		});
+	});
+
+	describe('getKey', () => {
+		const data1 = { key: 'myKey1', value: 'myValue' };
+		const data2 = { key: 'myKey2', value: 'myValue' };
+
+		beforeEach(async () => {
+			await ChainMetaEntity.setKey(data1.key, data1.value);
+			await ChainMetaEntity.setKey(data2.key, data2.value);
+		});
+
+		it('should resolve with error when invoked without key', async () => {
+			expect(ChainMetaEntity.getKey()).rejects.toThrow(
+				'Must provide the key to get'
+			);
+		});
+
+		it('should return matching result value with provided filters', async () => {
+			expect(await ChainMetaEntity.getKey(data1.key)).toEqual(data1.value);
+		});
+
+		it('should resolve with null if provided filter does not match', async () => {
+			expect(await ChainMetaEntity.getKey('custom-key')).toBeNull();
+		});
+	});
+
+	describe('setKey', () => {
+		it('should resolve with error when invoked without key', async () => {
+			expect(ChainMetaEntity.setKey()).rejects.toThrow(
+				'Must provide the key to set'
+			);
+		});
+
+		it('should resolve with error when invoked without value', async () => {
+			expect(ChainMetaEntity.setKey('myKey')).rejects.toThrow(
+				'Must provide the value to set'
+			);
+		});
+
+		it('should create key value pair if not exists', async () => {
+			const key = 'myKey';
+			const value = 'myValue';
+			await ChainMetaEntity.setKey(key, value);
+
+			const result = await getAllMeta(ChainMetaEntity.adapter);
+
+			expect(result).toEqual([{ key, value }]);
+		});
+
+		it('should update the value if key already exists', async () => {
+			const key = 'myKey';
+			const value = 'myValue';
+			const updatedValue = 'myUpdatedValue';
+
+			await ChainMetaEntity.setKey(key, value);
+			await ChainMetaEntity.setKey(key, updatedValue);
+
+			const result = await getAllMeta(ChainMetaEntity.adapter);
+			expect(result).toEqual([{ key, value: updatedValue }]);
+		});
+	});
+
+	describe('delete', () => {
+		const data1 = { key: 'myKey1', value: 'myValue' };
+		const data2 = { key: 'myKey2', value: 'myValue' };
+
+		beforeEach(async () => {
+			await ChainMetaEntity.setKey(data1.key, data1.value);
+			await ChainMetaEntity.setKey(data2.key, data2.value);
+		});
+
+		it('should delete the matching key record from database', async () => {
+			// Before delete verify they exists
+			expect(await getAllMeta(ChainMetaEntity.adapter)).toEqual([data1, data2]);
+
+			await ChainMetaEntity.delete({ key: data1.key });
+
+			expect(await getAllMeta(ChainMetaEntity.adapter)).toEqual([data2]);
+		});
+
+		it('should not throw error if no matching record found', async () => {
+			const nonExistingKey = 'nonExistingKey';
+
+			await expect(
+				ChainMetaEntity.delete({ key: nonExistingKey })
```

Comment:
why not await inside here?

Reply:
We want to check if returned promise resolved to null, that's one way to do it in Jest. Other case is to resolve to value first and match later.